app/views/assignments/_form.html.erb

```diff
@@ -489,4 +489,22 @@
                    data: { disable_with: t('working') } %>
     </p>
   <% end %>
+
+  <% if (action_name=="edit") %>
+    <% has_groups = @assignment.groups.length != 0 %>
+    <%= button_to t(:delete),
+                  course_assignment_path(@current_course, @assignment),
+                  {
+                    data: { confirm: "Are you sure you want to delete this assignment?" },
+                    method: 'delete',
+                    class: has_groups ? nil : 'button',
+                    id: has_groups ? nil : 'assignment-delete',
```

Comment:
Following my above comment, don't use an `id` here, instead conditionally add a second class to `'button'` above